package-res/resources/web/pentaho/type/_type.js

```diff
@@ -0,0 +1,1362 @@
+       *
+       * @return {boolean} Returns `true` if any attribute was added, `false`, otherwise.
+       *
+       * @protected
+       *
+       * @see pentaho.type.Instance#toSpecInContext
+       */
+      _fillSpecInContext: function(spec, keyArgs) {
+        var any = false;
+        var isJson = keyArgs.isJson;
+
+        if(this._labelSet && O.hasOwn(this, "_label")) {
+          any = true;
+          spec.label = this._label;
+        }
+
+        // Normal attributes
+        _normalAttrNames.forEach(function(name) {
+          var _name = "_" + name, v;
+          // !== undefined ensures refinement fields are well handled as well
+          if(O.hasOwn(this, _name) && (v = this[_name]) !== undefined) {
+            any = true;
+            spec[name] = v;
+          }
+        }, this);
+
+        // Custom attributes
+        if(!this.isRefinement && this.isAbstract) {
+          any = true;
+          spec.isAbstract = true;
+        }
+
+        var styleClass = this._styleClass;
+        if(styleClass) {
+          any = true;
+          spec.styleClass = styleClass;
+        }
+
+        var viewInfo = O.getOwn(this, "_view");
+        if(viewInfo !== undefined) { // can be null
+          var view = viewInfo && viewInfo.value;
+          if(!view || !isJson || !F.is(view)) {
+            any = true;
+            spec.view = view;
+          }
+        }
+
+        return any;
+      },
+
+      /**
+       * Returns a _reference_ to this type.
+       *
+       * This method returns a reference to this type that is appropriate
+       * to be the value of an [inline type]{@link pentaho.type.spec.IInstance#_} property
+       * that is included on a specification of an instance of this type.
+       *
+       * If an [ambient specification context]{@link pentaho.type.SpecificationContext.current},
+       * currently exists, it is used to manage the serialization process.
+       * Otherwise, one is created and set as current.
+       *
+       * When a type is not anonymous, the cheapest reference,
+       * the [id]{@link pentaho.type.Type#id}, is returned.
+       *
+       * For anonymous types, a [temporary]{@link pentaho.type.SpecificationContext.isIdTemporary},
+       * serialization-only id is generated.
+       * In the first occurrence in the given scope,
+       * that id is returned, within a full specification of the type,
+       * obtained by calling [toSpecInContext]{@link pentaho.type.Type#toSpecInContext}.
+       * In following occurrences, only the previously used temporary id is returned.
+       *
+       * Some standard types have a special reference syntax,
+       * e.g. [List.Type#toRef]{@link pentaho.type.List.Type#toRef}.
+       *
+       * @see pentaho.type.Type#toSpec
+       *
+       * @param {Object} [keyArgs] - The keyword arguments object.
+       * Passed to every type and instance serialized within this scope.
```

Comment:
Suggest rewording into a sentence structure:

The keyword arguments object is passed to every type and instance serialized within this scope.